tools/ci/verify_bundled_optional.sh

```diff
@@ -19,22 +19,16 @@
 
 ## Checks that all bundled dependencies are marked as optional in the poms
 MVN_CLEAN_COMPILE_OUT=$1
-CI_DIR=$2
-FLINK_ROOT=$3
 
-source "${CI_DIR}/maven-utils.sh"
+MVN=${MVN:-./mvnw}
 
-cd "$FLINK_ROOT" || exit
+dependency_plugin_output=/tmp/optional_dep.txt
 
-dependency_plugin_output=${CI_DIR}/optional_dep.txt
-
-run_mvn dependency:tree -B > "${dependency_plugin_output}"
+$MVN dependency:tree -B > "${dependency_plugin_output}"
 
 cat "${dependency_plugin_output}"
 
-cd "${CI_DIR}/flink-ci-tools/" || exit
-
-run_mvn exec:java -Dexec.mainClass=org.apache.flink.tools.ci.optional.ShadeOptionalChecker -Dexec.args=\""${MVN_CLEAN_COMPILE_OUT}" "${dependency_plugin_output}"\"
+$MVN -pl tools/ci/flink-ci-tools exec:java -Dexec.mainClass=org.apache.flink.tools.ci.optional.ShadeOptionalChecker -Dexec.args="${MVN_CLEAN_COMPILE_OUT} ${dependency_plugin_output}"
 EXIT_CODE=$?
 
 if [ $EXIT_CODE != 0 ]; then
```

Comment:
I'm wondering whether we should pipe the console output of `ShadeOptionalChecker` through `tee` and have the output available in a file. We could add the file path to the echo statement below (rather than referring to the "previous output for details". It's kind of annoying to scroll through the stdout logs

Reply:
We could also print the input parameters of the script to allow a copy&paste to trigger a rerun.

Reply:
> It's kind of annoying to scroll through the stdout logs

I think that's more of a symptom to be solved by sorting the output by severity (important stuff at the end). You shouldn't need to scroll at all to find the thing that broke CI.

Reply:
I ran it locally and it worked fine.
IIRC we don't check that the dependency-plugin call fails; in which case we pipe garbage into the checker which typically gives you this error.

Reply:
note that you can also do `tools/ci/compile.sh | tee output.txt`. The scripts doesn't necessarily have to take care of that.

Reply:
>  IIRC we don't check that the dependency-plugin call fails; in which case we pipe garbage into the checker which typically gives you this error.

added a check to prevent this from silently happening.